blocks/identity-block/components/login/index.jsx

```diff
@@ -48,16 +51,24 @@ const useLogin = ({
 			})
 
 			const fullURL = searchParams.get("redirect") ? appendURLParams(searchParams.get("redirect"), aditionalParams.filter(item => item !== undefined)) : null;
-			const validatedRedirectParam = validateURL(fullURL);
+			const validatedRedirectParam = validateURL(fullURL, contextPath);
 			setRedirectQueryParam(validatedRedirectParam);
 		}
 
 		if (redirectToPreviousPage && document?.referrer) {
-			const redirectUrl = new URL(document.referrer);
+			const redirectUrlLocation = new URL(document.referrer);
 
-			setRedirectToURL(`${redirectUrl.pathname}${redirectUrl.search}`);
+			if (searchParams.has('reset_password')) {
+				setCurrentRedirectToURL(`${redirectURL}${redirectUrlLocation.search}`);
+			} else {
+				const newRedirectUrl = redirectUrlLocation.pathname.includes('/pagebuilder/')
```

Comment:
Added this to handle when we are on the edit page in pagebuilder and we go to the preview page. Since it has a referrer it will redirect back to the edit page if we do not have this logic.

Reply:
`isAdmin` is already being passed to this hook. That should be used to prevent this redirect from happening in the first place. Something like `if (redirectToPreviousPage && document?.referrer && !isAdmin)`